crates/tlb/src/cell/pruned_branch_cell.rs

```diff
@@ -0,0 +1,100 @@
+use crate::cell::CellBehavior;
+use crate::cell::HigherHash;
+use crate::cell_type::CellType;
+use crate::level_mask::LevelMask;
+use crate::Cell;
+use bitvec::order::Msb0;
+use bitvec::prelude::BitVec;
+use sha2::{Digest, Sha256};
+use std::sync::Arc;
+
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub struct PrunedBranchCell {
+    pub data: BitVec<u8, Msb0>,
+}
+
+impl CellBehavior for PrunedBranchCell {
+    #[inline]
+    fn as_type(&self) -> CellType {
+        CellType::PrunedBranch
+    }
+
+    #[inline]
+    fn data(&self) -> &BitVec<u8, Msb0> {
+        self.data.as_ref()
+    }
+
+    #[inline]
+    fn references(&self) -> &[Arc<Cell>] {
+        &[]
+    }
+
+    #[inline]
+    fn level(&self) -> u8 {
+        self.data.as_raw_slice().first().cloned().unwrap()
+    }
+
+    #[inline]
+    fn max_depth(&self) -> u16 {
+        self.depths().into_iter().max().unwrap_or(0)
+    }
+}
+
+impl HigherHash for PrunedBranchCell {
+    #[inline]
+    fn level_mask(&self) -> LevelMask {
+        LevelMask::new(
+            self.data
+                .as_raw_slice()
+                .first()
+                .cloned()
+                .expect("invalid data length"),
+        )
+    }
+
+    #[inline]
+    fn higher_hash(&self, level: u8) -> [u8; 32] {
+        if self.level_mask().contains(level) {
+            self.data.as_raw_slice()[1 + (32 * level) as usize..1 + (32 * (level + 1)) as usize]
+                .try_into()
+                .expect("invalid data length")
```

Comment:
Avoid panics. There should be no assumptions on data that you get from untrusted sources